### PR TITLE
Mc 37 move element back to desk

### PR DESCRIPTION
**Pull request by danielrittrich**
- Info for Reviewers -
-----------------------

I have adapted the movement function from Simeon and now it works without any problems and the components running back. At the same time, I also removed the "old / WIP" movement.

But what I was missing, was the selling station on the main branch and to prevent any further merge conflicts, I merged the main into the MC-37 to straighten everything out there first.

--> Therefore the info: New changes have now only happened there:

- ComponentHandler.cs
- ComponentMovement.cs
- ComponentSpawner.cs

### CODE REVIEW

**Comment by danielrittrich** [comment]
By the way, I made a bugfix for the IsEqual() function in the case and mb. we had a major logic error in there.

File: Merge Conflict/Assets/Scripts/ComponentMovement.cs [hunk 1]
```diff
@@ -17,8 +17,10 @@ public class ComponentMovement : MonoBehaviour
     public float MoveMinDistance = 30;
     public float MoveMaxDistance = 70;
     public float MoveSpeed = 50;
+    public float ReturningMovingSpeed = 2000;
     private readonly float _smoothMovementFactor = 0.02f;
     // Distance by which the component should move during the current movement
+    private Vector2 CenterOfDesk = new(600f, 900f);
```
Comment (Xipit) [inline Merge Conflict/Assets/Scripts/ComponentMovement.cs:23]: CenterOfDesk should be calculated at start by using the RectTransform of the desk.

**Review by Xipit — COMMENTED** [review]
1 small change, then you can merge on your own

File: Merge Conflict/Assets/Scripts/ComponentMovement.cs (hunk 1 above)
Comment (danielrittrich) [reply Merge Conflict/Assets/Scripts/ComponentMovement.cs]: is geupdatet.

**Pull request merged**